### PR TITLE
CP-41-create-the-media-component-to-display-tuits-that-have-media-content

### PR DESCRIPTION
**Pull request by kunchao322**
Added media component which uses service.findTuitsWithMediaByUser(uid)

### CODE REVIEW

File: src/components/profile/media.js [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+        service.findTuitsWithMediaByUser("my")
+            .then(tuits => setMediaTuits(tuits));
+    useEffect(findMyMediaTuits, []);
     return(
-        <div>
-            <h1>Media</h1>
-        </div>
+        <>
+            <h1>Tuits with media should be displayed below</h1>
+            <Tuits tuits={mediaTuits}
```
Comment (nilzc) [inline src/components/profile/media.js:13]: I think this should work. But can you also take a look at ```Tuits``` component? I believe now we have ```image``` and ```video``` fields, and how to display them is not taken care of yet.

Comment (nilzc) [reply src/components/profile/media.js:13]: I created 2 tuits with media under account "lin", password is "haha". You can test rendering the 2 tuits and verify the media is displayed.

**Review by nilzc — CHANGES_REQUESTED** [review]

File: src/components/profile/media.js [hunk 2]
```diff
@@ -1,8 +1,21 @@
-const Media = () => {
+import {useEffect, useState} from "react";
+import * as service from "../../services/tuits-service";
+import Tuits from "../tuits";
+
+const MyMedia = () => {
```
Comment (nilzc) [inline src/components/profile/media.js:5]: Why change ```Media``` to ```MyMedia```?

Comment (kunchao322) [reply src/components/profile/media.js:5]: To render under profile after logged in

File: src/components/profile/media.js (hunk 1 above)
Comment (kunchao322) [reply src/components/profile/media.js:13]: tuit.js is fixed to render images and videos

File: src/components/profile/media.js [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+    const [mediaTuits, setMediaTuits] = useState([]);
+    const findMyMediaTuits = () =>
+        service.findTuitsWithMediaByUser("my")
+            .then(tuits => setMediaTuits(tuits));
+    useEffect(findMyMediaTuits, []);
     return(
-        <div>
-            <h1>Media</h1>
-        </div>
+        <>
```
Comment (nilzc) [inline src/components/profile/media.js]: You can remove ```<></>``` and directly return Tuits component

File: src/components/tuits/tuit-image.js [hunk 4]
```diff
@@ -1,17 +1,10 @@
 import React from "react";
-const TuitImage = ({tuit}) => {
+const TuitImage = ({image}) => {
+    console.log(image);
```
Comment (nilzc) [inline src/components/tuits/tuit-image.js]: remove ```console.log()```

File: src/components/tuits/tuit-images.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import TuitImage from "./tuit-image";
+
+const TuitImages = ({tuit}) => {
+    const images = tuit.image || [];
+    console.log(images);
+    console.log(images);
+  return(
+      <>
```
Comment (nilzc) [inline src/components/tuits/tuit-images.js]: remove ```<></>```

File: src/components/tuits/tuit-images.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import TuitImage from "./tuit-image";
+
+const TuitImages = ({tuit}) => {
+    const images = tuit.image || [];
+    console.log(images);
+    console.log(images);
```
Comment (nilzc) [inline src/components/tuits/tuit-images.js]: remove ```console.log()```

File: src/components/tuits/tuit-video.js [hunk 7]
```diff
@@ -1,16 +1,19 @@
...
-      className="ttr-responsive-video ttr-rounded-15px position-relative overflow-hidden w-100 mt-2">
-      <iframe width="560"
-              height="315" src={tuit.video}
-              title="YouTube video player"
-              frameBorder="0"
-              className="position-absolute top-0 w-100 h-100"
-              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
-              allowFullScreen></iframe>
-    </div>
+      <>
```
Comment (nilzc) [inline src/components/tuits/tuit-video.js]: remove ```<></>```

**Review by nilzc — APPROVED** [review]

**Pull request merged**